source/FreeRTOS_IP_Utils.c

```diff
@@ -1151,7 +1151,7 @@ const char * FreeRTOS_strerror_r( BaseType_t xErrnum,
             /* MISRA Ref 21.6.1 [snprintf and logging] */
 /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-216 */
             /* coverity[misra_c_2012_rule_21_6_violation] */
-            ( void ) snprintf( pcBuffer, uxLength, "Errno %d", ( int ) xErrnum );
+            ( void ) snprintf( pcBuffer, uxLength, "Errno %d", ( int32_t ) xErrnum );
```

Comment:
```suggestion
            ( void ) snprintf( pcBuffer, uxLength, "Errno %d", ( int ) xErrnum );
```

Reply:
"int" is the violation:  Using basic numerical type "int" rather than a typedef that includes size and signedness information.

Reply:
"%d" printf strings match an int which is almost always the type of BaseType_t.

int32_t -> 32 bits
int on avr / BaseType_t -> 16 bits
int on arm / BaseType_t -> 32 bits

Thus no cast is necessary.

Reply:
That makes sense.  I will remove the cast

Reply:
@paulbartell 

> Thus no cast is necessary.

I'm afraid that without the cast many compilers ( except 64-bit ) will complain. The `%d` can not be used in combination with `int32_t` or `BaseType_t`, even when they have the same size as a natural `int`.

I wrote "guidelines" about the use of formats and casts [as a comment on PR197](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/197#pullrequestreview-595091959).
~~~
    int int_value;
    int32_t int32_value;
    BaseType_t base_value;
    printf( "int_value = %d\n", int_value );
    printf( "int32_value = %d\n", int32_value );  // Compiler warning
    printf( "base_value = %d\n", base_value );    // Compiler warning
~~~
@xuelix's PR is already merged, When MISRA is ( almost ) ready, I will check it once more.